frontend/rust-lib/flowy-search/src/folder/handler.rs

```diff
@@ -0,0 +1,28 @@
+use flowy_error::FlowyResult;
+
+use crate::entities::SearchResultPB;
+use crate::services::indexer::IndexManager;
+use crate::services::manager::ISearchHandler;
+
+use super::indexer::FolderIndexManager;
+
+pub struct FolderSearchHandler {
+  index_manager: Box<FolderIndexManager>,
+}
+
+impl FolderSearchHandler {
+  pub fn new(index_manager: Box<FolderIndexManager>) -> Self {
+    Self { index_manager }
+  }
+}
+
+impl ISearchHandler for FolderSearchHandler {
```

Comment:
The naming convention for traits in Rust typically focuses on the behavior or characteristic they represent, without any specific prefix or suffix. So how about rename to `SearchHandler`